rfd/xxxx-bot-for-cert-renewals.md

```diff
@@ -0,0 +1,298 @@
+
+The primary user this agent is designed for is the admin who wants to leverage
+Teleport for as many systems as possible, but simply does not have the time to
+keep track of all certificates and manually renew them.
+
+While users with sophisticated automation and dedicated credential management
+tools may benefit from this solution, they likely already have the resources to 
+automate certificate renewals.
+
+As a result, a primary goal for this tool is simplicity. We will optimize the user
+experience for those looking to add systems to their cluster quickly and easily rather
+than supporting a multitude of configuration options and hooks for customization.
+
+### Overview
+
+We will develop a new lightweight tool, `tbot`, which runs as an agent on
+external systems that are part of a Teleport cluster (a VM running a Postgres
+database, or an OpenSSH server, for example).
+
+This agent will join the Teleport cluster, request certificates, store them in a
+user-specified [destination](#destination), and then monitor the certificates
+and renew them as necessary.
+
+#### User Experience
+
+First, a token will be generated for the bot to join the cluster.
+
+```
+$ tctl tokens add --type=bot
+The invite token: 13b74f49d27536dd5c514073097c197b
+This token will expire in 60 minutes
+
+...
+```
+
+Next, the agent will be started using the generated token (full set of options
+omitted here for brevity):
+
+```
+$ tbot start --name=foo --auth-server=proxy.example.com:3080 --token=TOKEN ...
+```
+
+Note that both direct dialing or connecting through a Teleport proxy are supported.
+
+The bot joins the cluster and can be seen with the `tctl` command:
+
+```
+$ tctl bots ls
+ID              NAME
+0123-4567       foo
+```
+
+TODO: what else might we want to see? public addr? number (and kind) of certs?
+
+In the event of an incident, the bot can be locked with the new `tctl lock`
+functionality. A locked bot will be unable to renew certificates and will have
+its existing certificates revoked. (TODO: this is how it works, right?)
+
+Note that a bot can manage multiple sets of certificates, and locking a
+particular bot will apply to *all* of the certificates it manages.
+
+```
+$ tctl lock --bot=0123-4567
+```
+
+The confirmation message when running a lock command can include a list of the
+certificates the bot is managing.
+
+```
+This bot is managing the following certificates:
+
+- x509 server certificate at /etc/ssl (expires in 7 days)
+- SSH client certificate at /etc/ssh (expires in 8 hours)
+
+Locking the bot will prevent the renewal of these certificates.
+Are you sure you want to continue? (y/n)
+```
+
+#### Renewals
+
+There are two scenarios under which the bot will initiate a renewal:
+
+1. When a certificate is nearing expiry. A certificate is considered near
+   expiration if 75% of its TTL has elapsed, or when there are 4 hours or less
+   until expiration (whichever is sooner).
+2. When a user and/or host CA rotation is taking place.
```

Comment:
Might want to have an option to force-rotate all certificates for testing or dry-run